securebanking-openbanking-uk-rcs-server/src/main/java/com/forgerock/securebanking/openbanking/uk/rcs/api/ConsentDetailsApiController.java

```diff
@@ -135,27 +135,22 @@ public ResponseEntity<ConsentDetails> getConsentDetails(String consentRequestJws
                 details.setUsername(consentClientRequest.getUser().getUserName());
                 details.setUserId(consentClientRequest.getUser().getId());
 
-                details.setAccounts(accountService.getAccountsWithBalance(details.getUserId()));
-
                 // DebtorAccount is optional, but the PISP could provide the account identifier details for the PSU
                 // The accounts displayed in the RCS ui needs to be The debtor account if is provided in the consent otherwise the user accounts
-                // TODO debtorAccount logic for VRP sweeping payments case until it will fixed for all payments  https://github.com/secureapigateway/secureapigateway/issues/731
-                if (details.getIntentType().equals(IntentType.DOMESTIC_VRP_PAYMENT_CONSENT)) {
-                    if(Objects.nonNull(((PaymentsConsentDetails) details).getDebtorAccount())) {
-                        setDebtorAccountWithBalance((PaymentsConsentDetails) details, consentRequestJws, intentId);
-                    }
+                if (Objects.nonNull(((PaymentsConsentDetails) details).getDebtorAccount())) {
```

Comment:
Is this cast safe?

Could details be an instance of AccountsConsentDetails at this point?

Reply:
Good point, is not a cast safe, I forget add the account details tests to reveal this kind of errors.
Fixed

Reply:
Great, thanks.

Is there a longer term goal to refactor RCS? As this controller in particular is getting a bit messy now.

This kind of if instance of then logic is quite procedural, there are better OOP ways of doing it. I think the controller is trying to do too much generically, when we really need controllers that specific to the intent type that they handle i.e. an AccountConsentDetailsController, PaymentsConsentDetailsController.

The ConsentDetailsApiController can be kept as the entry point, it can then do any generic work and dispatch to the applicable controller to handle the specific intent type.

This refactoring is non trivial so it's probably best raised as a new issue to be prioritised (or added as a comment any existing RCS refactoring issue). Then we can merge this PR and come back to address this later.

Reply:
Issue created to refactor RCS ConsentDetailsControler: https://github.com/SecureApiGateway/SecureApiGateway/issues/773